mlir/unittests/Interfaces/DataLayoutInterfacesTest.cpp

```diff
@@ -200,6 +205,15 @@ struct SingleQueryType
     return Attribute();
   }
 
+  Attribute getDefaultMemorySpace(DataLayoutEntryInterface entry) {
+    static bool executed = false;
```

Comment:
That's not thread safe/

Reply:
That was the existing mechanism:
https://github.com/llvm/llvm-project/blob/main/mlir/unittests/Interfaces/DataLayoutInterfacesTest.cpp#L203-L229

Should I change all of them?

Reply:
Oh it's just in a local method inside a unit-test, never mind then.